Sources/GRPCHTTP2Core/GRPCStreamStateMachine.swift

```diff
@@ -1211,9 +1211,11 @@ extension GRPCStreamStateMachine {
         return .rejectRPC(trailers: trailers)
       }
 
-      let path = headers.firstString(forKey: .path)
-        .flatMap { MethodDescriptor(fullyQualifiedMethod: $0) }
-      if path == nil {
+      guard
+        let path = headers.firstString(forKey: .path).flatMap({
+          MethodDescriptor(fullyQualifiedMethod: $0)
+        })
+      else {
```

Comment:
Can we split this into two checks?
1. We get a ":path", and 
2. The ":path" is a valid `MethodDescriptor`

Otherwise 1.) can be fine and 2.) not-fine and the error would indicate there was ":path" which is misleading